### PR TITLE
Implements video to virtual schedule veteran workflow in new component

### LINKED ISSUE #14894: Add hearing type dropdown to non-modal schedule veterans view
## Description

Add a hearing type dropdown to the non-modal schedule veterans view defined in #14892. When toggling the hearing type dropdown to virtual, fields for email notifications should display

## Acceptance criteria

- [ ] Add hearing type dropdown field to non-modal schedule veterans form
- [ ] Ensure that toggling to virtual shows the fields for email notifications
- [ ] Ensure that toggling back hides email notification fields
- [ ] Break out hearing conversion components into shareable groups

## Out of Scope

Hold off on connecting this to the backend work and updating the actions to display the new view

## Background/context/resources

Connected to #12941

Designs: https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=6022%3A21726&viewport=-269%2C32%2C0.06443598121404648&scaling=min-zoom

### PR DESCRIPTION
**Pull request by sahalliburton**
Resolves #14894, depends on merging #14938 and #14951 

### Description
Implemented the changes to the ScheduleVeteranForm component to include the virtual hearings state

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Run storybook locally: make storybook
1. Navigate to the schedule veteran component Hearings/Components/Schedule Veteran
1. Ensure that all scenarios match the [Figma diagram](https://www.figma.com/proto/V87TZArfdurCGJiEjQ73ES/Virtual-Hearings?node-id=6022%3A21726&viewport=-269%2C32%2C0.06443598121404648&scaling=min-zoom)

### CODE REVIEW

File: client/app/hearings/components/ScheduleVeteranForm.stories.js [hunk 1]
```diff
@@ -70,8 +70,20 @@ RegionalOfficeSelected.args = {
   }
 };
 
-export const WithErrors = Template.bind({});
-WithErrors.args = {
+export const VideoToVirtualConversion = Template.bind({});
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteranForm.stories.js:73]: Could you add some story component for scheduling central hearings to virtual?

File: client/app/reader/ReaderLoadingScreen.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { LOGO_COLORS } from '../constants/AppConstants';
 
 export class ReaderLoadingScreen extends React.Component {
   createLoadPromise = () => {
-    if (this.props.loadedAppealId && this.props.loadedAppealId === this.props.vacolsId) {
+    if (!this.props.loadedAppealId && this.props.loadedAppealId === this.props.vacolsId) {
```
Comment (rubaiyat22) [inline client/app/reader/ReaderLoadingScreen.jsx:15]: Is this supposed to be negated?

File: client/app/hearings/components/ScheduleVeteranForm.jsx [hunk 3]
```diff
@@ -28,65 +31,136 @@ export const ScheduleVeteranForm = ({
   const ro = appeal.regionalOffice || hearing.regionalOffice || initialRegionalOffice;
   const location = appeal.hearingLocation || hearing.location;
   const header = `Schedule ${appellantTitle} for a Hearing`;
+  const virtual = hearing?.virtualHearing;
+  const video = hearing?.readableRequestType === 'Video';
   const availableHearingLocations = orderBy(appeal.availableHearingLocations || [], ['distance'], ['asc']);
   const dynamic = ro !== appeal.closestRegionalOffice || isEmpty(appeal.availableHearingLocations);
 
+  const handleChange = () => {
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteranForm.jsx:39]: Is this meant to update the queue redux state? Will this work be completed in https://app.zenhub.com/workspaces/caseflow-5915dd178f67e20b5553ba0c/issues/department-of-veterans-affairs/caseflow/15029

File: client/app/reader/ReaderLoadingScreen.jsx (hunk 2 above)
Comment (sahalliburton) [reply client/app/reader/ReaderLoadingScreen.jsx:15]: No it is not, I am not sure how that got left in there so thanks for catching that!

File: client/app/hearings/components/ScheduleVeteranForm.jsx (hunk 3 above)
Comment (sahalliburton) [reply client/app/hearings/components/ScheduleVeteranForm.jsx:39]: Yes this is intended to integrate with the redux state later and will be completed in that later ticket

File: client/app/hearings/components/ScheduleVeteranForm.jsx [hunk 4]
```diff
@@ -28,65 +31,136 @@ export const ScheduleVeteranForm = ({
...
-                onChange={(hearingLocation) => props.onChange('appeal', { hearingLocation })}
-                dynamic={dynamic}
-                staticHearingLocations={availableHearingLocations}
-              />
+        {virtual ? (
+          <React.Fragment>
+
+            <div className="usa-width-one-half" >
+              <ReadOnly spacing={0} label="Regional Office" text={appeal.regionalOffice} />
+              <ReadOnly spacing={15} label="Hearing Location" text={appeal.hearingLocation?.name} />
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteranForm.jsx:66]: The location is supposed to be `Virtual` if the the hearing is being converted to virtual from video or central. below is an image of figma diagram
Screen Shot 2020-08-25 at 2 56 09 PM

File: client/app/hearings/components/ScheduleVeteranForm.stories.js [hunk 5]
```diff
@@ -70,8 +70,31 @@ RegionalOfficeSelected.args = {
...
+VideoToVirtualConversion.args = {
+  appeal: {
+    ...amaAppeal,
+    regionalOffice: defaultHearing.regionalOfficeKey,
+    hearingLocation: scheduleHearingDetails.hearingLocation }
+};
+
+export const CentralToVirtualConversion = Template.bind({});
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteranForm.stories.js:81]: For some reason the story component isn't showing veteran timezone field

Screen Shot 2020-08-25 at 3 03 54 PM

File: client/app/hearings/components/ScheduleVeteranForm.jsx (hunk 4 above)
Comment (sahalliburton) [reply client/app/hearings/components/ScheduleVeteranForm.jsx]: Thanks I updated this

File: client/app/hearings/components/ScheduleVeteranForm.stories.js (hunk 5 above)
Comment (sahalliburton) [reply client/app/hearings/components/ScheduleVeteranForm.stories.js:81]: It looks like there were some props not getting passed down, thanks for catching this!

File: client/app/hearings/components/ScheduleVeteranForm.jsx [hunk 6]
```diff
@@ -28,65 +32,137 @@ export const ScheduleVeteranForm = ({
...
-                onChange={(hearingLocation) => props.onChange('appeal', { hearingLocation })}
-                dynamic={dynamic}
-                staticHearingLocations={availableHearingLocations}
-              />
+        {virtual ? (
+          <React.Fragment>
+
+            <div className="usa-width-one-half" >
+              <ReadOnly spacing={0} label="Regional Office" text={appeal.regionalOffice || 'Central'} />
+              <ReadOnly spacing={15} label="Hearing Location" text={appeal.hearingLocation?.name || 'Virtual'} />
```
Comment (rubaiyat22) [inline client/app/hearings/components/ScheduleVeteranForm.jsx:67]: the readonly location should always be `Virtual` in conversion to virtual form

**Pull request merged**